nexus/src/authn/external/spoof.rs

```diff
@@ -293,8 +302,12 @@ mod test {
         // It currently returns `NotRequested`.  The specific code isn't
         // important so much as that we don't crash or succeed.
         let test_header = make_header_value_raw(b"foo\x80ar").unwrap();
+        let mut map = http::HeaderMap::new();
+        map.insert(&http::header::AUTHORIZATION, test_header);
+        let typed_header = map.typed_get::<Authorization<Bearer>>();
+        assert!(typed_header.is_none());
```

Comment:
This seems to mostly be testing the `http` library now. If you get rid of that part you're left with

```rs
assert!(matches!(authn_spoof(None), SchemeResult::NotRequested));
```

which you already have a test for.

Reply:
Maybe a more constructive version of the point: this test would make more sense if `authn_spoof` took the whole `HeaderMap`, that way you'd be testing what we do with `HeaderMap` rather than what `HeaderMap` does in itself.

Reply:
demo: https://github.com/oxidecomputer/omicron/compare/authorization-header...auth-header-use-map

Reply:
I hear what you're saying here.  But I wanted one slightly more end-to-end test for this case (i.e., instead of relying on `http` and `headers` working correctly).  I agree once we hit L308, L309-L312 are redundant with another test, but it seemed worthwhile for the clarity and double-check that it really produces what we think it should.  Arguably, we already have more complete end-to-end tests that actually go through the HTTP server, but another one here doesn't seem to hurt.

The change to have `authn_mode` take a `HeaderMap` seems fine, but I slightly prefer that it take only the header that it reads.  It seems to me that it ought not to have a bunch of other information that's unrelated to what it does.  This isn't a strong preference but given what's already here I'm planning to leave it.

Thanks for the review!